### PR TITLE
misc: replace black with ruff formatter.

### PR DESCRIPTION
**Pull request by JDarDagran**
### Problem

ruff formatter is lightning fast and replaces black in >99% cases.

### Solution

Adjust the code to missing rules.

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: integration/airflow/openlineage/airflow/adapter.py [hunk 1]
```diff
@@ -277,22 +266,21 @@ def _build_run(
...
                 runId=parent_run_id,
                 namespace=_DAG_NAMESPACE,
                 name=parent_name,
             )
-            facets.update({
-                "parent": parent_run_facet,
-                "parentRun": parent_run_facet,  # Keep sending this for the backward compatibility
-            })
+            facets.update(
+                {
```
Comment (mobuchowski) [inline integration/airflow/openlineage/airflow/adapter.py:278]: I hate this style :(
Can we change it?

File: client/python/openlineage/client/run.py [hunk 2]
```diff
@@ -34,7 +33,7 @@ class Dataset(RedactMixin):
     name: str = attr.ib()
     facets: Dict[Any, Any] = attr.ib(factory=dict)
 
-    _skip_redact: List[str] = ["namespace", "name"]
+    _skip_redact: ClassVar[List[str]] = ["namespace", "name"]
```
Comment (mobuchowski) [inline client/python/openlineage/client/run.py:36]: Nice to automatically detect and hint this.

**Review by mobuchowski — COMMENTED** [review]
Overall great work @JDarDagran, just a few questions.

File: integration/airflow/openlineage/airflow/adapter.py (hunk 1 above)
Comment (JDarDagran) [reply integration/airflow/openlineage/airflow/adapter.py:278]: I didn't find any format options as such.
May I ask what's wrong with this style? :)

File: ruff.toml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+line-length = 110
+target-version = "py38"
+exclude = [
+    "integration/spark"
```
[the remark this answers is not in the document]
Comment (JDarDagran) [reply ruff.toml:4]: There are pyspark scripts that ruff does not play well, e.g. `spark` variable is added to context but there's no way (AFAIK) to simulate this in ruff.
Flink doesn't have any Python scripts, therefore there's no need to filter that out.

File: client/python/openlineage/client/run.py (hunk 2 above)
Comment (JDarDagran) [reply client/python/openlineage/client/run.py:36]: Not sure if I understand this comment properly.
Are you highlighting it as nice feature or some additional action to do?

File: integration/airflow/openlineage/airflow/adapter.py (hunk 1 above)
Comment (JDarDagran) [reply integration/airflow/openlineage/airflow/adapter.py:278]: Also: https://docs.astral.sh/ruff/formatter/black/

File: client/python/openlineage/client/run.py (hunk 2 above)
Comment (JDarDagran) [reply client/python/openlineage/client/run.py:36]: https://github.com/OpenLineage/OpenLineage/pull/2260/commits/9efdc644f1c1b3dfd2308643428fe5eedc6312ac
seems to do the trick

File: ruff.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (JDarDagran) [reply ruff.toml:4]: https://github.com/OpenLineage/OpenLineage/pull/2260/commits/9efdc644f1c1b3dfd2308643428fe5eedc6312ac
seems to do the trick

**Comment by pawel-big-lebowski** [comment]
This is dynamite 💣

File: client/python/openlineage/client/run.py (hunk 2 above)
Comment (mobuchowski) [reply client/python/openlineage/client/run.py:36]: Yes, I mean that's a nice feature 🙂

File: integration/airflow/openlineage/airflow/adapter.py (hunk 1 above)
Comment (mobuchowski) [reply integration/airflow/openlineage/airflow/adapter.py:278]: Nothing, just that my eyes hurt. However, I believe pretty much _any_ sane enforced formatting is better than lack of it, so if we can't change it I'm okay with that.

File: ruff.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (JDarDagran) [reply ruff.toml]: I've added it in `.pre-commit-config.yaml`.

**Pull request merged**